File: server/router/campus/campus_report.go
package campus

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type CampusReportRouter struct{}

func (r *CampusReportRouter) InitCampusReportRouter(Router *gin.RouterGroup) {
	reportRouter := Router.Group("campusReport").Use(middleware.OperationRecord())
	reportRouterWithoutRecord := Router.Group("campusReport")
	{
		reportRouter.POST("handleCampusReport", campusReportApi.HandleCampusReport)
	}
	{
		reportRouterWithoutRecord.GET("getCampusReportList", campusReportApi.GetCampusReportList)
		reportRouterWithoutRecord.GET("findCampusReport", campusReportApi.FindCampusReport)
	}
}

File: web/src/api/campusReport.js
import service from '@/utils/request'

export const getCampusReportList = (params) => {
  return service({
    url: '/campusReport/getCampusReportList',
    method: 'get',
    params
  })
}

export const findCampusReport = (params) => {
  return service({
    url: '/campusReport/findCampusReport',
    method: 'get',
    params
  })
}

export const handleCampusReport = (data) => {
  return service({
    url: '/campusReport/handleCampusReport',
    method: 'post',
    data
  })
}

